### PR TITLE
Function wrappers for creators 

### PR DESCRIPTION
**Pull request by mdsumner**
Implements https://github.com/dcooley/sfheaders/issues/63

Adds functions `sf_pt(), sf_mpt(), sf_line(), sf_mline(), sf_poly(), sf_mpoly()` that wrap the core functions `sf_point(), sf_linestring(), sf_multilinestring(), sf_polygon(), sf_multipolygon()`.  

* These expect a data frame with with standard required column names as appropriate for each geometry type. 
* Columns `z` and / or `m` column names are optional, and used if present. 
* New functions use `close  = FALSE` as default. 
 

Main documentation is in `?sf_pt()` under section **Helpers**, and this rationale section is copied across the other functions, as per the section-copy style already in use. 

(I haven't incorporated https://github.com/dcooley/sfheaders/tree/issues/64 or checked it, I'd prefer to have these merged and then check from there?)

### CODE REVIEW

**Comment by SymbolixAU** [comment]
> New functions use close = FALSE as default

May I ask why you chose this as the default?

> I haven't incorporated https://github.com/dcooley/sfheaders/tree/issues/64 or checked it, I'd prefer to have these merged and then check from there?

yeah that's fine.

**Comment by mdsumner** [comment]
The `close` thing is definitely hanging there undecided, I consider checking for closed-polygons to be a slow thing - and my 90% use-case is I'm working with tables that already have that assumed. It seemed a natural complement to "I have a data frame with these standard column names expected by this function" -  so of course they are closed. 

If there's no real overhead to the close-logic then I would change that. (checking now)

**Comment by mdsumner** [comment]
Wow, it definitely doesn't seem to take much time - so I'm happy to change. 

- [ ] set close = TRUE
- [ ] improve sanity checks and error message (not stopifnot)

File: R/sf-helpers.R [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+#' (sfx <- sf_pt( x, keep = TRUE ))
+#'
+#' ## we trivially round-trip with sf_pt()
+#' sf_pt(sf_to_df(sfx, fill = TRUE), keep = TRUE)
+#' @export
+sf_pt <- function(obj, keep = FALSE) {
+  check_columns(obj, c("x", "y"), "sf_pt")
+  call_args <- name_matcher(obj)
+  call_args[c("obj", "keep")] <- list(obj, keep)
+  ## needs rethinking
```
Comment (SymbolixAU) [inline R/sf-helpers.R]: @mdsumner is this still in the "thinking" stage, or are you happy with this?

**Comment by SymbolixAU** [comment]
It also just occurred to me that the use-case for this is exactly what I had in mind for users working at the C++ level. You can round-trip from `Rcpp` code without specifying column names. So thanks for bringing this up a level to R.

File: R/sf-helpers.R (hunk 1 above)
Comment (mdsumner) [reply R/sf-helpers.R]: I think it's fine, it's a bit ugly is all!

**Comment by mdsumner** [comment]
@dcooley just to confirm, if we use `close = TRUE` sfheaders *checks* if closing is required yeah, and only does  that if needed?

**Comment by SymbolixAU** [comment]
correct.

Implementation is [here](https://github.com/dcooley/sfheaders/blob/master/inst/include/sfheaders/sfg/polygon/close_polygon.hpp). 

- if `close = FALSE` the `close_polygon()` function is exited early
- if `close = TRUE` && `first_row != last_row`, it adds the first row to the end of the matrix.

**Comment by mdsumner** [comment]
cool thanks, I think I found an example where it fails so confirming that is helpful

**Comment by mdsumner** [comment]
false alarm, it's a bug in my PR for 'to_sf', which I'll withdraw because you have a grander solution on your radar ;)

**Pull request merged**